### PR TITLE
Align networkmanager_dns_mode to the RHEL 9 STIG

### LINKED ISSUE #11702: networkmanager_dns_mode is misaligned with DISA
#### Description of problem:
networkmanager_dns_mode is misaligned with DISA

#### Details:

DISA rule probably fails because of spaces around `=` added by our Ansible remediation.

#### Outcome:
SSG result: pass
DISA result: fail

The issue is present in these test variants:
- ansible

#### SCAP Security Guide Version:
Current upstream master as of 2024-03-12 as of HEAD cbbca44.

####  External Content's Version:
DISA STIG RHEL 9 V1R1

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

Removes the ability to add space around the `dns=none` per the STIG SCAP content.

#### Rationale:

Fixes #11702

### CODE REVIEW

**Comment by jan-cerny** [comment]
The problem is that the Ansible remediation still adds or doesn't remove the spaces. Ansible has an option that could be used https://docs.ansible.com/ansible/latest/collections/community/general/ini_file_module.html#parameter-no_extra_spaces. OTOH Bash remediation is OK.

**Comment by Mab879** [comment]
> The problem is that the Ansible remediation still adds or doesn't remove the spaces. Ansible has an option that could be used https://docs.ansible.com/ansible/latest/collections/community/general/ini_file_module.html#parameter-no_extra_spaces. OTOH Bash remediation is OK.

I have updated the Ansible remediation.

File: shared/macros/10-ansible.jinja [hunk 1]
```diff
@@ -733,6 +733,7 @@ The following macro remediates Audit syscall rule in :code:`/etc/audit/audit.rul
     value: "{{{ value }}}"
     create: yes
     mode: 0644
+    ignore_spaces: false
```
Comment (jan-cerny) [inline shared/macros/10-ansible.jinja:736]: The tests still fail.

If we do this they should stop failing:

```suggestion
    ignore_spaces: false
    no_extra_spaces: true
```

**Review by jan-cerny — CHANGES_REQUESTED** [review]

File: linux_os/guide/system/network/networkmanager/networkmanager_dns_mode/ansible/shared.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 {{{ ansible_instantiate_variables("var_networkmanager_dns_mode") }}}
 
-{{{ ansible_ini_file_set("/etc/NetworkManager/NetworkManager.conf", "main", "dns", "{{ var_networkmanager_dns_mode }}") }}}
+{{{ ansible_ini_file_set("/etc/NetworkManager/NetworkManager.conf", "main", "dns", "{{ var_networkmanager_dns_mode }}", true) }}}
```
Comment (jan-cerny) [inline linux_os/guide/system/network/networkmanager/networkmanager_dns_mode/ansible/shared.yml:9]: I'm sorry I was wrong, this isn't sufficient, they still fail. We need also to update the macro here.

```suggestion
{{{ ansible_ini_file_set("/etc/NetworkManager/NetworkManager.conf", "main", "dns", "{{ var_networkmanager_dns_mode }}", description=None, ignore_spaces=True) }}}

```

**Review by jan-cerny — CHANGES_REQUESTED** [review]

**Review by jan-cerny — APPROVED** [review]
The Automatus CI jobs fail because it isn't possible to start the networkmanager service in a container environment that we use in the CI. I have executed the tests locally using a RHEL 9 virtual machine backed and they all passed:

```
jcerny@fedora:~/work/git/scap-security-guide (pr/11745)$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel9  networkmanager_dns_mode
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2024-03-27-1640/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_networkmanager_dns_mode
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script correct_default.pass.sh using profile (all) OK
INFO - Script missing.fail.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
INFO - Script space.fail.sh using profile (all) OK
jcerny@fedora:~/work/git/scap-security-guide (pr/11745)$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel9 --remediate-using ansible  networkmanager_dns_mode
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2024-03-27-1642/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_networkmanager_dns_mode
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script correct_default.pass.sh using profile (all) OK
INFO - Script missing.fail.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
INFO - Script space.fail.sh using profile (all) OK
```

**Pull request merged**